Libraries/LibURL/Pattern/Init.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2025, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Optional.h>
+#include <AK/String.h>
+
+namespace URL::Pattern {
+
+// https://urlpattern.spec.whatwg.org/#dictdef-urlpatterninit
+struct Init {
+    Optional<String> protocol;
+    Optional<String> username;
```

Comment:
The only thing about putting it in LibURL is it will add an additional dependency on it to LibRegex, but I think that's fine